src/Entity/ProgramManager.php

```diff
@@ -307,6 +300,80 @@ public function addProgram(AddProgramRequest $request): ?Program
     return $program;
   }
 
+  /**
+   * Adds a new program from a scratch_program. Doesn't add the Project file.
+   */
+  public function createProgramFromScratch(?Program $program, User $user, array $program_data): ?Program
+  {
+    $modified_time = TimeUtils::dateTimeFromScratch($program_data['history']['modified']);
+    if (null === $program)
+    {
+      $program = new Program();
+      $program->setUser($user);
+      $program->setScratchId($program_data['id']);
+      $program->setDebugBuild(false);
+    }
+    else
+    {
+      if ($program->getLastModifiedAt() === $modified_time)
+      {
+        return $program;
+      }
+      $program->incrementVersion();
+    }
+    $program->setCatrobatVersion(1);
```

Comment:
I'm not sure about this version number.

Reply:
All usages of setCatrobatVersion have 1 as parameter. Maybe we should delete that column?
The info about the catrobat version seems to be in CatrobatVersionName.
Should that one be set even if no project file currently exists?

Reply:
Yeah, for sure! But it can be done in a different ticket. It would be necessary to rework this version feature anyway. Instead of the string should use the typical major.minor.patch format. The sql string comparison right now is a little bit hacky - requiring exactly the correct format (same string length) to work. **Ticket needed**

Reply:
No if there is no project it should not be set.

Reply:
Added [SHARE-333](https://jira.catrob.at/browse/SHARE-333)